### ISSUE TITLE
Cross-Compilation using mingw fails on package-config check due to file ending

### ISSUE THREAD

**Issue by jeremiahpslewis** [valhalla/valhalla#3931]
Hey! As part of building Valhalla for Julia, I ran into the following compilation error building for windows:

The path `install(FILES ${VALHALLA_BINARY_DIR}/libvalhalla.pc` should be ` install(FILES ${VALHALLA_BINARY_DIR}/libvalhalla.pc.exe`. The patch that makes Windows compile is here, although presumably it will need to be special cased so that other platforms still build. 

https://github.com/JuliaPackaging/Yggdrasil/blob/a6fed5cad25617e78902ed925490b1f2477279c2/V/Valhalla/bundled/patches/pkg-config-mingw.patch#L9

**Comment by nilsnolde** [comment]
Thanks, don't think anyone every tried to actually _install_ on Windows:) We had a working MinGW64 build to at least make sure it'll build, but Fedora Rawhide proved to be too unstable. What's the build env you're using? 

Maybe you can generally show us a bit more what it is you're doing? We might want to include it in https://github.com/valhalla/valhalla#related-projects.

To your problem though: if you want to PR a stable fix that works for all platforms we'd be happy to include it, and could make sure it's working via CI.

**Comment by jeremiahpslewis** [comment]
Build environment is this one: https://binarybuilder.org which ships binaries for the Julia programming language.

**Comment by jeremiahpslewis** [comment]
I'm not a CMake wiz, but I gave it a try :)

**Comment by jeremiahpslewis** [comment]
@nilsnolde Hmm, the more I look into this, the more it seems like the [pkg-config section of the CMake file](https://github.com/valhalla/valhalla/blob/f7c31e5c232dd20156235ef1ba2184dc5a27309a/src/CMakeLists.txt#L284) may not be very cross-platform friendly.

I have no experience with pkg-config, so only based on some googling, I was wondering...

would it to be possible to rewrite it using something like `pkg_check_modules(OPENCV REQUIRED IMPORTED_TARGET opencv)`?

**Comment by nilsnolde** [comment]
Sorry @jeremiahpslewis been a bit busy lately. I have not much experience with MinGW builds and the complications it entails. Let's leave this open for anyone who has the time and will to solve it. (wouldn't wait for it though;))

**Comment by nilsnolde** [comment]
Seems like you merged into master of the Julia repo anyways? So it wasn't necessary in the end or you'll live without Win builds?

Feel free to add your project/bindings to this repo's README.

**Comment by jeremiahpslewis** [comment]
Yep, dropped the windows builds for the moment. I'm also not building many of the optional features, will probably expand this first, then consider (maybe) tackling windows.
